kivy/uix/actionbar.py

```diff
@@ -486,24 +486,39 @@ def on_use_separator(self, instance, value):
             group.use_separator = value
         self.overflow_group.use_separator = value
 
+    def remove_widget(self, widget):
+        super(ActionView, self).remove_widget(widget)
+        if isinstance(widget, ActionOverflow):
+            for item in widget.list_action_item:
+                self._list_action_items.remove(item)
+
+        if widget in self._list_action_items:
+            self._list_action_items.remove(widget)
+
     def _clear_all(self):
+        lst = []
+        for item in self._list_action_items:
+            lst.append(item)
         self.clear_widgets()
         for group in self._list_action_group:
             group.clear_widgets()
 
         self.overflow_group.clear_widgets()
         self.overflow_group.list_action_item = []
+        self._list_action_items = lst
 
     def _layout_all(self):
         # all the items can fit to the view, so expand everything
         super_add = super(ActionView, self).add_widget
         self._state = 'all'
         self._clear_all()
-        super_add(self.action_previous)
+        if self.action_previous.parent is None:
+            super_add(self.action_previous)
         if len(self._list_action_items) > 1:
-            for child in self._list_action_items[1:]:
-                child.inside_group = False
-                super_add(child)
+            for child in self._list_action_items[:]:
```

Comment:
Yeah this is not required anymore. I added this because while making  changes the list items were disarranged a little..I just checked right now it works fine without this change.